Framework/CurveFitting/src/Algorithms/SplineSmoothing.cpp

```diff
@@ -167,13 +167,15 @@ void SplineSmoothing::performAdditionalFitting(MatrixWorkspace_sptr ws,
  *it with the desired number of spectra.
  * Also labels the axis of each spectra with Yi, where i is the index
  *
- * @param inws :: The input workspace
+ * @param inws :: The input workspace as a shared pointer
  * @param size :: The number of spectra the workspace should be created with
  * @return The pointer to the newly created workspace
  */
 API::MatrixWorkspace_sptr
 SplineSmoothing::setupOutputWorkspace(const MatrixWorkspace_sptr &inws,
                                       const int size) const {
+  // Must pass a shared pointer instead of a reference as the
+  // workspace factory will not accept raw pointers.
```

Comment:
Hah yes... I have an open pull request that adds an improved factory for workspaces, so soon we will be able to fix this, but no need to do it now.